packages/neuron-wallet/src/utils/queue.ts

```diff
@@ -0,0 +1,49 @@
+import { AsyncQueue, AsyncResultIterator, queue } from 'async'
+
+export default function queueWrapper<T, R, E = Error>(
+  fn: (item: T) => Promise<R>,
+  concurrency?: number,
+  ignoreSameItem?: boolean
+): AsyncQueue<T> & { asyncPush: (item: T) => Promise<R | undefined> } {
+  const itemList: T[] = []
+  const promiseList: Promise<R | undefined>[] = []
+  const queueFn: AsyncResultIterator<T, R, E> = (item: T, callback: (err?: E | null, res?: R) => void) => {
+    try {
+      fn(item)
+        .then((v: R) => {
+          callback(null, v)
+        })
+        .catch((err: E) => {
+          callback(err)
+        })
+    } catch (error) {
+      callback(error)
+    }
+  }
+  const q = queue<T, R, E>(queueFn, concurrency)
+  return Object.defineProperty(q, 'asyncPush', {
+    value: (item: T) => {
+      if (ignoreSameItem && itemList.length) {
+        if (item === itemList[itemList.length - 1] && q.length()) {
```

Comment:
Is shallow comparison enough in our cases?

Reply:
Yeah, it's used to avoid reset sync task many times in a short time like debounce. Most of time, we don't need to ignore the same item.